src/current/v23.2/column-level-encryption.md

```diff
@@ -0,0 +1,327 @@
+{{site.data.alerts.end}}
+
+## Security considerations
+
+### Whether to use `encrypt` and `decrypt` or `encrypt_iv` and `decrypt_iv`?
+
+Both `encrypt` and `decrypt` have `*_iv` variants: `encrypt_iv` and `decrypt_iv`.
+
+The benefits of using `encrypt_iv` and `decrypt_iv` include:
+
+- Avoid Repetition: If you use the same encryption key and method to encrypt the same plaintext multiple times without an IV, you'll get the same ciphertext every time. This repetition can provide a point of attack for someone trying to break the encryption. By using an IV, even the same plaintext will produce different ciphertexts, provided a different IV is used each time.
+- Defend Against Pattern Analysis: Without an IV, if two users have the same SSN or piece of data, their encrypted values will also be the same. An attacker can exploit these patterns. By using different IVs for each encryption, the encrypted values will be different even if the plaintext values are the same.
+- Cipher Block Chaining (CBC) Mode: Many encryption algorithms, like AES, operate on blocks of data. In modes like CBC, the previous block of ciphertext is used as an IV for the encryption of the next block. This means that even if there are patterns in the plaintext, they won't appear in the ciphertext. However, for the first block, there is no previous block of ciphertext, so an IV is used. This is another way IVs help in breaking up patterns in the ciphertext.
+- Mitigate Replay Attacks: Since the IV is typically random and changed for every encryption, it makes replay attacks more difficult. An attacker can't simply take an old piece of encrypted data and send it again, as the IV will likely have changed.
+
+The drawbacks of using `encrypt_iv` and `decrypt_iv` include:
+
+- Storage: You need to store the IV alongside the ciphertext. It's common practice to prepend or append the IV to the ciphertext before storing it. Unlike the encryption key, the IV doesn't need to be kept secret, but it does need to be known for decryption.
+- Randomness: It's crucial that IVs are random and not predictable. If an attacker can predict the next IV, some of the security benefits are negated.
+- Unique IVs with the Same Key: While IVs need to be random, it's also essential that the same IV isn't used twice with the same encryption key. Doing so can leak information about the plaintext.
+
+You will need to assess your risk profile to determine which functions to use.
+
+### How AES variants are determined
+
+The actual AES variant (AES-128, AES-192, or AES-256) is determined by the length of the encryption key you provide in the functions:
+
+- AES-128: 16-byte key
+- AES-192: 24-byte key
+- AES-256: 32-byte key
+
+Keys have to be 16-, 24-, or 32-byte lengths and map to the corresponding AES encryption strength.
```

Comment:
The user is allowed to use a different length key, the key is then padded up to the next valid key length or truncated if it is over 32 bytes long.

Reply:
oh thanks for this

updated to say

> Internally, keys have to be 16-, 24-, or 32-byte lengths and map to the corresponding AES encryption strength. If you use a key with a different length, the key is padded up to the next valid key length. If the key is over 32 bytes long, it is truncated.

please let me know if you see any issue with that language

Reply:
lgtm! some minor suggestions:
* "padded" -> "zero-padded"
* "truncated" -> "truncated to 32 bytes"

Reply:
added the suggested text! thanks